src/ActionPage.php

```diff
@@ -159,4 +165,27 @@ public function register_post_meta() {
 			register_post_meta( self::POST_TYPE, $field, $args );
 		}
 	}
+
+	/**
+	 * On load of permalinks page, add a action slug setting field.
+	 */
+	public function p4_load_permalinks() {
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing
+		if ( isset( $_POST['p4_action_posttype_slug'] ) ) {
+			update_option( 'p4_action_posttype_slug', sanitize_title_with_dashes( $_POST['p4_action_posttype_slug'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		}
+
+		// Add a settings field to the permalink page.
+		add_settings_field( 'p4_action_posttype_slug', __( 'Action post type slug', 'planet4-master-theme-backend' ), [ $this, 'add_action_slug_field' ], 'permalink', 'optional' );
+	}
+
+	/**
+	 * Add Action slug text field on permalinks page.
+	 */
+	public function add_action_slug_field() {
+
+		$value = get_option( 'p4_action_posttype_slug' );
+		echo '<input type="text" value="' . esc_attr( $value ) . '" name="p4_action_posttype_slug" id="p4_action_posttype_slug" class="regular-text" /><p>' . esc_html__( 'The default Action post type slug is "action".', 'planet4-master-theme-backend' ) . '</p>';
```

Comment:
Just curious, why do you use `_` instead of `-` to the `name` and `id` fields?

Reply:
Nothing specific, I use the same name (`p4_action_posttype_slug`) for wp option setting key hence  got added for name & id, but if we need we can change it.

Reply:
It's just to follow the same rule of our templates, but not required to change it